lib/Conch.pm

```diff
@@ -96,11 +96,10 @@ sub startup {
             # header is specified and greater than 0). Content-Type header must
             # be specified and must be 'application/json' for all payloads, or
             # HTTP status code 415 'Unsupported Media Type' is returned.
-            if ($headers->content_length) {
-                unless ($headers->content_type
-                    && $headers->content_type =~ /application\/json/i) {
-                    return $c->status(415);
-                }
+            if ($headers->content_length
+                and (not $headers->content_type
+                     or $headers->content_type !~ /application\/json/i)) {
+                return $c->status(415);
```

Comment:
So of the changes in this commit, this is the only one that honestly seems clearer to me than the original, and I'm not 100% convinced it is (speaking as someone trying to follow the change to make sure the logic is identical).